src/main/kotlin/net/thesilkminer/mc/ematter/common/feature/sbg/SbgTileEntity.kt

```diff
@@ -0,0 +1,200 @@
+package net.thesilkminer.mc.ematter.common.feature.sbg
+
+import net.minecraft.block.Block
+import net.minecraft.nbt.NBTTagCompound
+import net.minecraft.tileentity.TileEntity
+import net.minecraft.util.EnumFacing
+import net.minecraft.util.ITickable
+import net.minecraftforge.common.capabilities.Capability
+import net.thesilkminer.kotlin.commons.lang.uncheckedCast
+import net.thesilkminer.mc.boson.api.direction.Direction
+import net.thesilkminer.mc.boson.api.energy.Holder
+import net.thesilkminer.mc.boson.api.energy.Producer
+import net.thesilkminer.mc.boson.prefab.energy.holderCapability
+import net.thesilkminer.mc.boson.prefab.energy.producerCapability
+import net.thesilkminer.mc.ematter.common.shared.TemperatureSources
+import kotlin.math.roundToInt
+
+@Suppress( "EXPERIMENTAL_API_USAGE" , "EXPERIMENTAL_OVERRIDE" , "EXPERIMENTAL_UNSIGNED_LITERALS" )
+class SbgTileEntity : TileEntity(), Producer, Holder, ITickable {
+
+    internal companion object {
+        private const val ROOM_TEMPERATURE: Int = 295
+        //that's the storage of the basic mad
+        private const val MAX_CAPACITY: ULong = 3_141uL
+
+        //next one is here in case you want to make this changeable through a config file in the future
+        private const val POWER_MULTIPLIER: Double = 1.0
```

Comment:
This should be placed in a configuration file, which isn't set up yet
Consider this on hold

Reply:
So, should I just remove the val or leave it for now?

Reply:
It's fine for now: this is more of a reminder for the future

Reply:
ok. I'll mark that with a comment on that line